packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -298,8 +298,9 @@ Future<void> injectPlugins(FlutterProject project) async {
   await _writeAndroidPluginRegistrant(project, plugins);
   await _writeIOSPluginRegistrant(project, plugins);
   if (!project.isModule && project.ios.hostAppRoot.existsSync()) {
+    final IosProject iosProject = IosProject.fromFlutter(project);
     final CocoaPods cocoaPods = CocoaPods();
-    if (plugins.isNotEmpty)
+    if (plugins.isNotEmpty || iosProject.podfile.existsSync())
```

Comment:
I don't think we should make both paths do the same thing given the semantics of the method. We're not necessarily trying to 'setup' the pod file. We're just integrating its dependency in the xcode project. 

I would instead add a public `addPodsDependencyToFlutterXcconfig(IosProject iosProject)` to `CocoaPods`. 

`else if` here to check if both the podfile and the Pods directory (implying the user at least ran pod install once) exist. If so, call CocoaPods.addPodsDependencyToFlutterXcconfig.

Then have CocoaPods.setupPodfile use addPodsDependencyToFlutterXcconfig. 

We should have `addPodsDependencyToFlutterXcconfig` call `_addPodsDependencyToFlutterXcconfig(IosProject iosProject, String mode)` for all the xcconfigs.